modules/core/src/test/java/org/apache/ignite/platform/PlatformCacheEntryEventFilterCpp.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.platform;
+
+import org.apache.ignite.Ignite;
+import org.apache.ignite.binary.BinaryObject;
+import org.apache.ignite.cache.CacheEntryEventSerializableFilter;
+import org.apache.ignite.resources.IgniteInstanceResource;
+
+import javax.cache.event.CacheEntryEvent;
+import javax.cache.event.CacheEntryListenerException;
+import java.util.UUID;
+
+/**
+ * Test filter.
+ */
+@SuppressWarnings({"FieldCanBeLocal", "FloatingPointEquality", "MismatchedReadAndWriteOfArray", "unused"
+})
+public class PlatformCacheEntryEventFilterCpp implements CacheEntryEventSerializableFilter {
+    /** Property to be set from platform. */
+    private String startsWith = "-";
+
+    /** Property to be set from platform. */
+    private char charField;
+
+    /** Property to be set from platform. */
+    private byte byteField;
+
+    /** Property to be set from platform. */
+    private short shortField;
+
+    /** Property to be set from platform. */
+    private int intField;
+
+    /** Property to be set from platform. */
+    private long longField;
+
+    /** Property to be set from platform. */
+    private float floatField;
+
+    /** Property to be set from platform. */
+    private double doubleField;
+
+    /** Property to be set from platform. */
+    private boolean boolField;
+
+    /** Property to be set from platform. */
+    private UUID guidField;
+
+    /** Property to be set from platform. */
+    private BinaryObject objField;
+
+    /** Property to be set from platform. */
+    private char[] charArr;
+
+    /** Property to be set from platform. */
+    private byte[] byteArr;
+
+    /** Property to be set from platform. */
+    private short[] shortArr;
+
+    /** Property to be set from platform. */
+    private int[] intArr;
+
+    /** Property to be set from platform. */
+    private long[] longArr;
+
+    /** Injected instance. */
+    @IgniteInstanceResource
+    private Ignite ignite;
+
+    /** {@inheritDoc} */
+    @Override public boolean evaluate(CacheEntryEvent event) throws CacheEntryListenerException {
```

Comment:
I''m totally puzzled about this one. Perhaps a small comment on how this is used would help?

Reply:
Well, yeah, this is used in tests to check that object properties passed from C++ to Java layer properly.